### PR TITLE
project: use isolated go to build pebble

### PR DESCRIPTION
**Pull request by sergiusens**
This allows a rockcrafter to install whatever version of go they would required.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Comment by sergiusens** [comment]
This is a PoC to fix #128

**Comment by cjdcordeiro** [comment]
interesting

File: tests/unit/test_project.py [hunk 1]
```diff
@@ -84,12 +84,16 @@ def yaml_loaded_data():
...
     return {
         "pebble": {
-            "plugin": "go",
+            "plugin": "nil",
             "source": "https://github.com/canonical/pebble.git",
-            "build-snaps": ["go/1.19/stable"],
+            "override-pull": (
+                "craftctl default\n"
+                "snap set system experimental.parallel-instances=true\n"
+                "snap install go_pebble --channel 1.19/stable --classic\n"
```
Comment (tigarmo) [inline tests/unit/test_project.py:92]: I see, so `go_pebble` means an instance of the `go` snap called `pebble`?
I found this reference, for whoever else was unaware for parallel instances: https://snapcraft.io/docs/parallel-installs

**Review by tigarmo — APPROVED** [review]
I know this is a draft but I say we merge it and let it play out a little bit, to see if any issues crop up

**Comment by sergiusens** [comment]
I would like for parallel installs to be natively supported by craft parts but also have some for of detection to not need to declare the equivalent of CC and CXX for go

**Comment by sergiusens** [comment]
none of what I mentioned is super important for this _hidden_ part until we start thinking about manifests

**Pull request merged**